internal/store/version.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"context"
+
+	"github.com/google/go-github/v38/github"
+)
+
+//go:generate mockgen -destination=../mocks/mock_version.go -package=mocks github.com/mongodb/mongocli/internal/store VersionDescriber
+
+type VersionDescriber interface {
+	LatestVersion() (string, error)
+}
+
+// LatestVersion encapsulates the logic to manage different cloud providers.
+func (s *Store) LatestVersion() (string, error) {
+	client := github.NewClient(nil)
+	release, _, err := client.Repositories.GetLatestRelease(context.Background(), "mongodb", "mongocli")
```

Comment:
I think there's a `config.ToolName` we can use here instead of `mongocli`

Reply:
Not too sure if we should, this is a repo name, toolname can be changed if we ever change the cli binary name, no?

Reply:
yeah, that's why I'm more on the maybe than must, just putting it out there theres a constant with the name, not sure what would happen if we decide to rename